src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexNode.cs

```diff
@@ -2622,5 +2126,89 @@ internal SymbolicRegexNode<TSet> PruneAnchors(uint prevKind, bool contWithWL, bo
+                    return _lower;
+
+                case SymbolicRegexNodeKind.Alternate:
+                    Debug.Assert(_left is not null && _right is not null);
+                    if (_left.IsNullableFor(context))
+                    {
+                        // Left is nullable, so the match is from the left
+                        return _left.ResolveFixedLength(context);
+                    }
+                    else
+                    {
+                        // Otherwise right must be nullable and thus the relevant match
+                        Debug.Assert(_right.IsNullableFor(context));
+                        return _right.ResolveFixedLength(context);
+                    }
+
+                case SymbolicRegexNodeKind.Concat:
+                    Debug.Assert(_left is not null && _right is not null);
+                    int leftLength = _left.ResolveFixedLength(context);
+                    return leftLength >= 0 ? leftLength : _right.ResolveFixedLength(context);
+            }
+            return -1;
+        }
+
+        /// <summary>
+        /// Break up a top level alternation into its elements. This is used when transitioning from DFA mode to NFA mode.
+        /// A <see cref="SymbolicRegexNodeKind.DisableBacktrackingSimulation"/> node on the top level will be unwrapped
+        /// and the resulting elements re-wrapped to maintain the metadata.
+        /// </summary>
+        /// <returns>an enumeration of the elements of the alternation, or just the node itself if there is no alternation</returns>
+        internal IEnumerable<SymbolicRegexNode<TSet>> BreakUpAlternation()
+        {
+            switch (_kind)
+            {
+                case SymbolicRegexNodeKind.DisableBacktrackingSimulation:
+                    Debug.Assert(_left is not null);
+                    // This call should never recurse more than one level
+                    Debug.Assert(_left._kind is not SymbolicRegexNodeKind.DisableBacktrackingSimulation);
+                    foreach (SymbolicRegexNode<TSet> element in _left.BreakUpAlternation())
+                    {
+                        // Re-wrap the element nodes in DisableBacktrackingSimulation if the top level node was too
+                        yield return _builder.CreateDisableBacktrackingSimulation(element);
+                    }
+                    break;
+                case SymbolicRegexNodeKind.Alternate:
+                    // Loop through all the elements of an alternation
+                    SymbolicRegexNode<TSet> current = this;
+                    while (current._kind is SymbolicRegexNodeKind.Alternate)
+                    {
+                        Debug.Assert(current._left is not null && current._right is not null);
+                        Debug.Assert(current._left._kind is not SymbolicRegexNodeKind.Alternate);
+                        // Alternations are in right associative form, so the left child is always an element of the alternation
```

Comment:
"so the left child is always an element of the alternation"... and is not itself an alternation, yes?

Reply:
That's a better way to put it, yes.